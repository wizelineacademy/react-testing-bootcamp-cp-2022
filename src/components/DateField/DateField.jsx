import styled from "styled-components";

const DateField = ({ date, setDate, handleClick }) => {
	const handleDateChange = (e) => {
		setDate(e.target.value);
	};

	return (
		<DateFieldContainer>
			<DatePicker
				type="date"
				value={date}
				data-testid="date-field"
				onChange={handleDateChange}
			/>
			<Button onClick={() => handleClick(date)}>Show</Button>
		</DateFieldContainer>
	);
};

const DateFieldContainer = styled.div`
	display: flex;
	justify-content: center;
`;

const DatePicker = styled.input`
	padding: 5px;
`;

const Button = styled.button`
	padding: 0px 10px;
	margin-left: 10px;
	border: 2px solid #b5e1b1;
	border-radius: 5px;
	background-color: #f0fbf1;
	&:hover {
		background-color: #e4f1e5;
	}
`;

export default DateField;
