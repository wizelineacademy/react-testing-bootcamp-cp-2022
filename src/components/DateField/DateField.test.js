import { render, screen } from "@testing-library/react";

import DateField from ".";

const setUp = (props) => render(<DateField {...props} />);

describe("DateField", () => {
	it("should render the date field and button", () => {
		const { container } = setUp();

		const dateField = screen.getByTestId("date-field");
		const button = screen.getByText(/show/i);

		expect(dateField).toBeInTheDocument();
		expect(button).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	describe("when the date prop is passed", () => {
		it("should display the date value", () => {
			setUp({ date: "2022-01-01" });

			const dateField = screen.getByTestId("date-field");

			expect(dateField).toHaveValue("2022-01-01");
		});
	});
});
