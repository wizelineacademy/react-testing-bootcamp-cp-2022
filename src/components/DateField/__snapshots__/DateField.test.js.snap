// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateField should render the date field and button 1`] = `
<div>
  <div
    class="sc-bczRLJ iIyXGc"
  >
    <input
      class="sc-gsnTZi firLOs"
      data-testid="date-field"
      type="date"
      value=""
    />
    <button
      class="sc-dkzDqf iBZyTi"
    >
      Show
    </button>
  </div>
</div>
`;
